core/domain/story_fetchers.py

```diff
@@ -465,13 +465,15 @@ def get_user_progress_in_story_chapters(
         if node.exploration_id
     ]
     exp_id_to_exp_map = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+    user_id_exp_id_combinations = list(itertools.product([user_id], exp_ids))
     exp_user_data_models = (
         user_models.ExplorationUserDataModel.get_multi(
-            [user_id], exp_ids))
+            user_id_exp_id_combinations))
 
     all_chapters_progress: List[
         story_domain.StoryChapterProgressSummaryDict] = []
-    for i, exp_id in enumerate(exp_ids):
+    for i, user_id_exp_id_pair in enumerate(user_id_exp_id_combinations):
```

Comment:
`for i, (user_id, exp_id) in ...`

(then I think you don't need the next line)

Reply:
I added `user_id_exp_id_pair` for avoiding to add lint ignore as user_id variable is not used later

Reply:
Ah then you can do `for i, (_, exp_id) in ...`

Reply:
Though what you have is fine too, so don't worry too much about this.